packages/gestalt/src/Typeahead.js

```diff
@@ -0,0 +1,351 @@
+  //     // TODO: Fix scrolling calcultation
+  //     // if (selected !== null && containerRef && selectedOptionRef)
+  //     // eslint-disable-next-line no-use-before-define
+  //     // scrollIntoView(containerRef, selectedOptionRef);
+  //   }, 100);
+
+  //   // eslint-disable-next-line react-hooks/exhaustive-deps
+  // }, [focused, selectedOptionRef]);
+
+  const handleFocus = () => {
+    // Internally set focus status
+    if (componentRef.current)
+      setFocused(componentRef.current.contains(document.activeElement));
+
+    // Run focus callback
+    if (onFocus) onFocus();
+  };
+
+  const handleBlur = () => {
+    // Clear input and reset options
+    if (options.length === 0) {
+      setSearch('');
+      setOptions(dataRef.current);
+    }
+
+    // TODO: Is this the best way to hide the results on blur
+    if (document.activeElement && componentRef.current) {
+      setTimeout(() => {
+        setFocused(document.activeElement === componentRef.current);
+      }, 200);
+    }
+    // Run blur callback
+    if (onBlur) onBlur();
+  };
+
+  // Handler for when text is typed
+  // This rule is stupid
+  // eslint-disable-next-line no-shadow
+  const handleChange = ({ syntheticEvent: event, value }) => {
+    // Filter the available options using original data
+    const updatedOptions = filterOriginalData(value);
+
+    // Update the avalble options
+    setOptions(updatedOptions);
+
+    // Update the search value
+    setSearch(value);
+
+    // Run onChange callback
+    if (onChange) onChange({ event, value });
+  };
+
+  const handleClear = () => {
+    setSelected(null);
+    setSearch('');
+    setFocused(false);
+  };
+
+  // Handler for when an item is clicked
+  const handleSelect = item => {
+    setSelected(item);
+
+    setSearch(item[searchField]);
+    setFocused(false);
+    if (onSelect) onSelect(item);
+  };
+
+  return (
+    <Box ref={componentRef}>
+      {/* INPUT FIELD */}
+      <InputField
+        accessibilityLabel={accessibilityLabel}
+        id={id}
+        value={search}
+        placeholder={placeholder}
+        size={size}
+        onChange={handleChange}
+        onFocus={handleFocus}
+        onBlur={handleBlur}
+        onClear={handleClear}
+        ref={inputRef}
+      />
+
+      {/* RESULTS CONTAINER */}
+      {focused && (
+        <Layer>
+          <Flyout
+            showCaret={caret}
+            anchor={inputRef.current}
+            idealDirection="down"
+            onDismiss={() => {}}
+            positionRelativeToAnchor={false}
+            size="flexible"
+          >
+            <Box
+              // ref={getContainerRef}
+              position="relative"
+              display="block"
+              padding={1}
+              marginTop={2}
+              marginBottom={2}
+              maxHeight={resultHeight}
+              width={`${inputRef?.current?.offsetWidth || 300 - 10}px`}
+              overflow="auto"
+              color={backgroundColor}
```

Comment:
@simkessy as mentioned before - let's remove these color options for the first release